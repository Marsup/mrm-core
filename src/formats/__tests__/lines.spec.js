'use strict';

jest.mock('fs');

const fs = require('fs');
const lines = require('../lines');

const data = [
	'one',
	'two',
];

fs.writeFileSync('test.lines', data.join('\n'));

it('should return an API', () => {
	const file = lines('notfound');
	expect(file).toEqual(expect.objectContaining({
		get: expect.any(Function),
		append: expect.any(Function),
		save: expect.any(Function),
	}));
});

it('get() should return all lines', () => {
	const file = lines('test.lines');
	expect(file.get()).toEqual(data);
});

it('append() should add lines', () => {
	const file = lines('test.lines');
	file.append('three', 'four');
	expect(file.get('foo')).toEqual([
		'one',
		'two',
		'three',
		'four',
	]);
});

it('append() should not reorder file when adding a line that already exists', () => {
	const file = lines('test.lines');
	file.append('two');
	expect(file.get('foo')).toEqual(data);
});

it('save() should create file', () => {
	const filename = 'new.lines';
	const file = lines(filename);
	file.append('foo', 'bar');
	file.save();
	expect(fs.readFileSync(filename, 'utf8')).toBe('foo\nbar');
});
