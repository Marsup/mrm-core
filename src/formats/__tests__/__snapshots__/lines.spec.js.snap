// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`save() should create file 1`] = `
Object {
  "/test.lines": "foo
bar",
}
`;

exports[`save() should save file without empty lines 1`] = `
Object {
  "/new.lines": "one
two
foo
bar",
}
`;
